src/metabase/driver/sql/query_processor.clj

```diff
@@ -44,9 +44,11 @@
   nil)
 
 (defn make-nestable-sql
-  "For embedding native sql queries, wraps [sql] in parens and removes any semicolons"
+  "For embedding native sql queries, wraps [sql] in parens and removes any comments and semicolons."
   [sql]
-  (str "(" (str/replace sql #";[\s;]*$" "") ")"))
+  (str "(" (-> sql
+               (str/replace #"--.*(\n|$)" "")
```

Comment:
This looks dangerous. It would throw away everything after `--` even if it occurs in a string or something similar. A smaller problem is that it might paste together two tokens that only have the comment and the newline between them. (This can be fixed by replacing with `"$1"`.

Reply:
I've changed the approach to comment/semi removal. I think it is a bit more robust now, but as @snoe pointed out proper solution would need the sql parsing. Please have a look.